### PR TITLE
Remove Robot2.0 code in grsim, replaced it with a transform from use_…

### PR DESCRIPTION
**Pull request by ThijsLuttikhuis**
added back in the 4 camera grsim,

change model used for grsim control

works pretty well ^tm

### CODE REVIEW

File: src/robot.cpp [hunk 1]
```diff
@@ -468,32 +468,34 @@ void Robot::incSpeed(int i,dReal v)
...
+    double deltaAngle = constrainAngle(vw - robotAngle);
+    if (deltaAngle < - M_PI)
+        deltaAngle += 2*M_PI;
+    else if (deltaAngle > M_PI )
+        deltaAngle -= 2*M_PI;
+
+    double angularVel = robotAngle - prevYaw;
+    prevYaw = robotAngle;
+
+    double kP = 5.0;
```
Comment (rolfvdhulst) [inline src/robot.cpp]: these values have not been thoroughly tested/tuned yet.

File: src/sslworld.cpp [hunk 2]
```diff
@@ -504,7 +504,7 @@ void SSLWorld::recvActions()
                     bool wheels = false;
                     if (packet.commands().robot_commands(i).has_wheelsspeed())
                     {
-                        if (packet.commands().robot_commands(i).wheelsspeed()==true)
+                        if (packet.commands().robot_commands(i).wheelsspeed())
```
Comment (rolfvdhulst) [inline src/sslworld.cpp:525]: I don't like this change, wheelspeed() is an auto generated function. This just checks if the field exists- not if it is false or true.

File: src/sslworld.cpp [hunk 3]
```diff
@@ -616,24 +616,25 @@ dReal normalizeAngle(dReal a)
 
 bool SSLWorld::visibleInCam(int id, double x, double y)
 {
-    id %= 4;
```
Comment (rolfvdhulst) [inline src/sslworld.cpp:637]: Please put the 4 camera's back in

File: src/sslworld.cpp [hunk 4]
```diff
@@ -815,9 +816,9 @@ void SSLWorld::sendVisionBuffer()
 {
     int t = timer->elapsed();
     sendQueue.push_back(new SendingPacket(generatePacket(0),t));
-    sendQueue.push_back(new SendingPacket(generatePacket(1),t+1));
```
Comment (rolfvdhulst) [inline src/sslworld.cpp:835]: same as above

**Review by rolfvdhulst — CHANGES_REQUESTED** [review]

File: src/sslworld.cpp (hunk 4 above)
Comment (ThijsLuttikhuis) [reply src/sslworld.cpp:835]: Yes, I'll fix

File: src/sslworld.cpp (hunk 2 above)
Comment (ThijsLuttikhuis) [reply src/sslworld.cpp:525]: Check, clang tidy was complaining I should remove ==true ;-)

I'll add it back in

File: src/robot.cpp (hunk 1 above)
Comment (ThijsLuttikhuis) [reply src/robot.cpp]: Tested it a little bit. kP<5 seems too slow and kD should not be too big.

These values could be compared to the speed of motordrivers 3.0 and changed accordingly later

File: src/sslworld.cpp (hunk 2 above)
Comment (ThijsLuttikhuis) [reply src/sslworld.cpp:525]: uhm I keep it in because it returns a bool

Comment (rolfvdhulst) [reply src/sslworld.cpp:525]: Yes but then also refactor the rest of the file, this just feels pointless if you're going to refactor one == true when there is many in the whole file.

Comment (rolfvdhulst) [reply src/sslworld.cpp:525]: anyways this doesn't matter too much, so I am approving, but If you're going to refactor you should do it on file-by-file  and not on a ' I saw these lines and c-lang complained' structure, in my opinion

**Review by rolfvdhulst — APPROVED** [review]

**Pull request merged**